### PR TITLE
Use crypt_keyfile_read to read key files

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Fixes: #471

This also unfortunately means bumping required version of cryptsetup to 2.3.0 which means libblockdev won't compile on older distributions like RHEL 7 (which also doesn't have libfdisk so we can't support it anyway), Debian 10 (which I assume wouldn't update to 3.0 anyway) and RHEL 8 (cryptsetup 2.3.0 was added in 8.2 which means our CentOS 8 machines won't work for now, but CentOS Stream is ok).

`crypt_keyfile_read` also needs `crypt_device` which made the code a little bit more complicated and added some duplication between the blob and non-blob functions but this will be fixed in the planned crypto plugin API cleanup (which I'm not doing now because it would break UDisks daily builds).

### CODE REVIEW

File: src/plugins/crypto.c [hunk 1]
```diff
@@ -1204,45 +1215,106 @@ gboolean bd_crypto_luks_add_key (const gchar *device, const gchar *pass, const g
...
+    gchar *msg = NULL;
+
+    msg = g_strdup_printf ("Started adding key to the LUKS device '%s'", device);
+    progress_id = bd_utils_report_started (msg);
+    g_free (msg);
 
     if (!pass && !key_file) {
         g_set_error (error, BD_CRYPTO_ERROR, BD_CRYPTO_ERROR_NO_KEY,
                      "No passphrase nor key file given, cannot add key.");
+        bd_utils_report_finished (progress_id, (*error)->message);
```
Comment (tbzatek) [inline src/plugins/crypto.c:1230]: Just thinking about adding `g_warn_if_fail (error != NULL)` since you now dereference `error`.

Comment (vojtechtrefny) [reply src/plugins/crypto.c:1234]: Lets deal with this in #527

**Comment by vojtechtrefny** [comment]
We discussed this with @vpodzime a little and even if don't 100 % agree with his point of view, it's probably not the best decision to require cryptsetup 2.3.0 right now when it's not available in major distributions. Lets wait for now, libblockdev 3.0 will take some more time and we can revisit this before the release.

**Comment by vojtechtrefny** [comment]
Replaced by #752

**Pull request closed without merging**